src/libraries/System.Private.Runtime.InteropServices.JavaScript/tests/System/Runtime/InteropServices/JavaScript/EnumMarshalTests.cs

```diff
@@ -0,0 +1,94 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Runtime.InteropServices.JavaScript;
+using System.Collections.Generic;
+using Xunit;
+
+namespace System.Runtime.InteropServices.JavaScript.Tests
+{
+    public static class EnumMarshalTests
+    {
+        public enum RequestCache
+        {
+            [EnumExport(EnumValue = ConvertEnum.Default)]
+            Default = -1,
+            [EnumExport("no-store")]
+            NoStore,
+            [EnumExport(EnumValue = ConvertEnum.ToUpper)]
+            Reload,
+            [EnumExport(EnumValue = ConvertEnum.ToLower)]
+            NoCache,
+            [EnumExport("force-cache")]
+            ForceCache,
+            OnlyIfCached = -3636,
+        }
+
+        public enum StyleColor : uint
+        {
+            RedNoExport,
+            [EnumExport(EnumValue = ConvertEnum.Default)]
+            Red = 0xff0000ff,
+            [EnumExport(EnumValue = ConvertEnum.Numeric)]
+            GreenNumeric = 0x00ff00ff,
+            [EnumExport("red", EnumValue = ConvertEnum.ToUpper)]
```

Comment:
Just wondering, is this a useful use case? Maybe we should have just two modes:
1. default - returns either `ContractName`, or the field name as a string
2. number - returns the numeric value of the enum

Reply:
Simplified no longer valid